library/src/main/java/com/datatorrent/lib/io/fs/AbstractFileInputOperator.java

```diff
@@ -124,6 +124,34 @@
   protected transient long skipCount = 0;
   private transient OperatorContext context;
 
+  /**
+   * The time between sending batch tuples. Default value is 1 hour. A batch
+   * interval could consist of multiple scan intervals
+   */
+  private long batchIntervalMillis = 60 * 60 * 1000; //1 hour
```

Comment:
Typically you have hourly input files or directories. Also, even if there was a real use case for this, it doesn't seem implemented correctly. There is no abstraction for triggering the batch tuple and it relies on system clock.